tests/contrib/names/test_names_datastreams.py

```diff
@@ -89,3 +125,57 @@ def test_orcid_http_reader(_, bytes_xml_entry):
     assert len(results) == 1
     assert bytes_xml_entry.entry == results[0].entry
     assert not results[0].errors
+
+
+def test_names_service_writer_create(app, expected_from_xml):
+    writer = NamesServiceWriter("rdm-names", system_identity)
+    record = writer.write(StreamEntry(expected_from_xml))
+    record = record.to_dict()
+
+    assert dict(record, **expected_from_xml) == record
+
+
+def test_names_service_writer_duplicate(app, expected_from_xml):
+    writer = NamesServiceWriter("rdm-names", system_identity)
+    _ = writer.write(stream_entry=StreamEntry(expected_from_xml))
+    with pytest.raises(WriterError) as err:
```

Comment:
Unless we _resolve_ every time (which is not expensive since it goes to ES), there is not DB deduplication because we use `recidv2` and we do not store PIDs for identifiers (metadata only).

Reply:
Fixed by implementing a customized `write` method.